javascript/javascript3/week1/cityWeather.js

```diff
@@ -0,0 +1,155 @@
+
+                // Display the data
+                // This data should be showed in your app:
+
+                // The chosen city
+                // Temperature
+                // Icon for the weather type
+                // Wind speed
+                // How clowdy it is
+                // When sunrise and sunset is
+                // Optional a map showing where the city is located
+
+function weatherToMyChoice(){
+const button = document.getElementById('weatherButton')
+button.addEventListener('click', inputChange)
+
+const WeatherLocationButton = document.getElementById('locationButton')
+WeatherLocationButton.addEventListener('click', weatherLocationCapture)
+
+const saveLocation = document.getElementById('saveLocation')
+saveLocation.addEventListener('click',saveToLocalhost)
+
+const body = document.querySelector('body')
+const div = document.createElement('div')
+body.appendChild(div)
+
+let newCity = document.getElementById('cityname')
+newCity.addEventListener('change',inputChange);
+retrieveFromLocalhost( fetchTheWeather)
+
+
+//    input textbox changed && weather button callback
+function inputChange(){
+    if (newCity.value){
+    fetchTheWeather(newCity.value);
+}
+else {
+  newCity.value = 'insert valid city name'
+}}
```

Comment:
Code indentation is very important, it helps other developer to read your code easily. so try format your code using tool like prettier or beautifier.